thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/business/service/UnsplashApiService.java

```diff
@@ -0,0 +1,69 @@
+package pl.edu.agh.quizzesthesis.business.service;
+
+import lombok.AllArgsConstructor;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClient.ResponseSpec;
+import pl.edu.agh.quizzesthesis.api.dto.PictureWithAuthorResponse;
+import pl.edu.agh.quizzesthesis.api.dto.TermPictureUpdateRequest;
+import pl.edu.agh.quizzesthesis.business.exception.UnsplashException;
+import pl.edu.agh.quizzesthesis.data.entity.Term;
+
+import static pl.edu.agh.quizzesthesis.Credentials.UNSPLASH_API_KEY;
+
+@Service
+@AllArgsConstructor
+public class UnsplashApiService {
+    private final TermService termService;
+    private final WebClient unsplashApiClient;
+    
+    public Term getPicture(Term term) {
```

Comment:
I would make this method `@Transactional`.

Reply:
Added `@Transactional` to method signature.